renderers/css/actions.py

```diff
@@ -36,6 +36,19 @@ class OpiExecuteCommand(OpiAction):
 class OpiOpen(OpiAction):
     """Renderer for write PV actions."""
     ACTION_TYPE = 'OPEN_DISPLAY'
+    MACRO_ERROR = 'Macros must have key and value as strings ({}:{} violates this rule)'
+
+    def render(self, actions_node, action_model):
+        action_node = super(OpiOpen, self).render(actions_node, action_model)
+        macros_node = et.SubElement(action_node, 'macros')
+        parent_macros_node = et.SubElement(macros_node, 'include_parent_macros')
+        parent_macros_node.text = 'true' if action_model._parent_macros else 'false'
+        for key, value in action_model._macros.items():
```

Comment:
the default for model._macros is None; this will raise an error if no macros have been set